### PR TITLE
ConnectionPool -> CassandraConnection

### PR DESCRIPTION
**Pull request by conorbros**
As per the discussion on https://github.com/shotover/shotover-proxy/pull/481

### CODE REVIEW

File: shotover-proxy/src/transforms/cassandra/connection.rs [hunk 1]
```diff
@@ -16,78 +16,41 @@ use tokio_util::codec::{FramedRead, FramedWrite};
...
+pub struct CassandraConnection<C: Codec> {
     host: String,
-    pub connections: Vec<UnboundedSender<Request>>,
+    pub connection: Option<UnboundedSender<Request>>,
     codec: C,
-    auth_func: fn(&OwnedUnorderedConnectionPool<C>, &mut UnboundedSender<Request>) -> Result<()>,
 }
 
-impl<C: Codec> fmt::Debug for OwnedUnorderedConnectionPool<C> {
+impl<C: Codec> fmt::Debug for CassandraConnection<C> {
```
Comment (rukai) [inline shotover-proxy/src/transforms/cassandra/connection.rs:25]: Can we use Derivative instead?

**Review by rukai — APPROVED** [review]
Are you planning to clear out more stuff?
I think theres still more that should be done, but as is this improves on a bunch of issues so happy to land as is with follow ups.

**Comment by conorbros** [comment]
Yeah, I want to improve the API (caller shouldn't need to provide the ID, `Request` struct leak the abstractions). As noted on the earlier PR we will want to wait for my changes to the multi-map library to land before merging. 

https://github.com/havarnov/multimap/pull/33

**Comment by conorbros** [comment]
Hmm seems to be stuck on the license bot again.

**Review by benbromhead — APPROVED** [review]
nice!

**Pull request merged**